docs/MatrixOne/Reference/Variable/system-variables/sql-mode.md

```diff
@@ -0,0 +1,202 @@
+
+## Why set `sql_mode`
+
+`sql_mode` can control the behavior of MatrixOne, including how to handle NULL values, perform insert operations, and sort and compare strings. It can ensure strict compliance with SQL standards and avoid non-standard behavior. In addition, `sql_mode` can help developers better identify errors and potential issues in SQL statements.
+
+## Default modes of `sql_mode`
+
+The following are the standard modes of `sql_mode`, which are also the default modes in MatrixOne:
+
+- `ONLY_FULL_GROUP_BY`: The `GROUP BY` clause is used to group query results and perform aggregate calculations on each group, such as `COUNT`, `SUM`, `AVG`, etc. In the `GROUP BY` clause, the specified columns are the grouping columns. Other columns can be identified in the `SELECT` list, including aggregate or non-aggregate function columns. Without the `ONLY_FULL_GROUP_BY` mode, if a non-aggregate function column is set in the `SELECT` list, MatrixOne will select any value that matches the `GROUP BY` column use it to calculate the aggregate function by default.
+
+   !!! note
+       If your table structure is complex and for ease of querying, you can disable the `ONLY_FULL_GROUP_BY` mode.
+
+- `STRICT_TRANS_TABLES`: When executing `INSERT` and `UPDATE` statements, an error will be reported if the data does not conform to the rules defined for the table.
+
+- `NO_ZERO_IN_DATE`: Prohibits inserting zero values into fields of type `DATE` or `DATETIME`.
+
+- `NO_ZERO_DATE`: Prohibits inserting or updating the field value of `0000-00-00` as a date or datetime type. The purpose of this mode is to avoid inserting invalid or illegal values into a date or datetime field and require the use of a valid date or datetime values. If such an operation is performed, an error will be reported. It should be noted that the `NO_ZERO_DATE` mode is only effective for insert or update operations. For existing `0000-00-00` values, they can still be queried and used.
+
+- `ERROR_FOR_DIVISION_BY_ZERO`: This mode throws an error when dividing by zero.
+
+- `NO_ENGINE_SUBSTITUTION`: This mode throws an error when executing `ALTER TABLE` or `CREATE TABLE` statements. The purpose of this mode is to force the use of the specified storage engine, preventing data inconsistencies or performance issues. The specified storage engine is unavailable or does not exist instead of automatically substituting it with another available storage engine. If automatic substitution of storage engines is desired, this mode can be removed from the sql_mode or set to other supported sql_mode methods. It is important to note that this mode only applies to `ALTER TABLE` or `CREATE TABLE` statements and does not affect the storage engine of existing tables.
+
+## Optional modes for sql_mode
```

Comment:
it appears that a lot of sql_mode don't exist at all in MatrixOne. Please make a full check.

Reply:
@YANGGMM Please make a full check.